nixos/modules/services/networking/sshx.nix

```diff
@@ -0,0 +1,157 @@
+{ config, lib, pkgs, ... }:
+let
+  cfg = config.services.sshx;
+  inherit (lib) types;
+in
+{
+  options.services.sshx = {
+    enable = lib.mkEnableOption (lib.mdDoc "sshx, a secure web-based, collaborative terminal.");
+
+    package = lib.mkPackageOptionMD pkgs "sshx-server" { };
+
+    nginx.enable = lib.mkEnableOption (lib.mdDoc ''
+      NGINX for virtual host management.
+      The virtual host will be named according to `services.sshx.settings.hostName`.
+      Further NGINX configuration can be done by adapting `services.nginx.virtualHosts.<name>`.
+      See [](#opt-services.nginx.virtualHosts) for further information.
+    '');
+
+    caddy.enable = lib.mkEnableOption (lib.mdDoc ''
+      Caddy for virtual host management.
+      The virtual host will be named according to `services.sshx.settings.hostName`.
+      Further Caddy configuration can be done by adapting `services.caddy.virtualHosts.<name>`.
+      See [](#opt-services.caddy.virtualHosts) for further information.
+    '');
+
+    settings = {
+      port = lib.mkOption {
+        type = types.port;
+        default = 8051;
+        example = 2222;
+        description = lib.mdDoc "The port for `sshx-server` to listen on.";
+      };
+
+      listenAddress = lib.mkOption {
+        type = types.str;
+        default = "127.0.0.1";
+        example = "0.0.0.0";
+        description = lib.mdDoc "The IP address or network interface for `sshx-server` to listen on.";
+      };
+
+      signingSecret = lib.mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/etc/sshx/secret";
+        description = lib.mdDoc ''
+          Secret used for signing session tokens.
+          The file must be in the format of 'SSHX_SECRET=<THE_ACTUAL_SECRET>'.
```

Comment:
Maybe replace this to use systemd's LoadCredential instead of an EnvironmentFile so the user can just specify the secret?
[example](https://github.com/NixOS/nixpkgs/blob/f8012b8090aea8c0acfd943d56460281565f9a14/nixos/modules/services/misc/cfdyndns.nix#L63)

Reply:
This requires sshx to be able to read a secret from a file, it does not currently.

Reply:
you could do it like cfdyndns and set the environment variable in a wrapper script

Reply:
I don't that benefits us much. I'd rather add support for reading from a file in sshx-server's source code.

Reply:
once that is added, that would be a breaking change, unless we do this "workaround" now

Reply:
I don't like the loadCredential + export variable hack. If you want I am willing to wait for a patch to the sshx-server src to add support for reading secret from a file.